package/MDAnalysis/core/groups.py

```diff
@@ -1745,6 +1748,9 @@ class SegmentGroup(GroupBase):
     SegmentGroups.
 
     """
+    def __repr__(self):
```

Comment:
I've got something that is sort of working, implemented in the `GroupBase` class. It's like this:
```python
def __str__(self):
    name = self.level.name
    if len(self) <= 10:
        return '<{}Group {}>'.format(name.capitalize(), repr(list(self.level)))
    else:
        return '<{}Group {}, ..., {}>'.format(name.capitalize(),
                                              repr(list(self.level)[:3])[:-1],
                                              repr(list(self.level)[-3:])[1:])
```
My guess is something like this is what you want:
```python
repr(rg)
>>> '<ResidueGroup with 10 residues>'
str(rg)
>>> <ResidueGroup [<Residue MET, 1>,
 <Residue ARG, 2>,
 <Residue ILE, 3>,
...,
 <Residue ALA, 8>,
 <Residue PRO, 9>,
 <Residue GLY, 10>]>
```
So that the `repr` from the individual elements that form the group are printed out.
But what would be the equivalent of `self.segments` or `self.residues` in the `GroupBase` class? Or is that not accessible from there, since `ResidueGroup`, `AtomGroup` and `SegmentGroup` inherit from it?